test/e2e_node/pod_conditions_test.go

```diff
@@ -134,6 +134,35 @@ func runPodFailingConditionsTest(f *framework.Framework, hasInitContainers, chec
 		// Verify PodReady is not set (since sandboxcreation is blocked)
 		_, err = getTransitionTimeForPodConditionWithStatus(p, v1.PodReady, false)
 		framework.ExpectNoError(err)
+
+		ginkgo.By("update pod related volume resource to unblock sandbox creation")
+
+		e2epod.NewPodClient(f).Update(ctx, p.Name, func(pod *v1.Pod) {
+			pod.Spec.Volumes = []v1.Volume{
+				{
+					Name: "secret",
+					VolumeSource: v1.VolumeSource{
+						Secret: &v1.SecretVolumeSource{
+							SecretName: "secret",
+						},
+					},
+				},
+			}
+			pod.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{
+				{
+					Name:      "secret",
+					MountPath: "/config",
+				},
+			}
+		})
+
+		// Verify ContainersReady is updated
+		_, err = getTransitionTimeForPodConditionWithStatus(p, v1.ContainersReady, true)
```

Comment:
I think you should add a check for `PodReadyToStartContainers` condition.

```
		if checkPodReadyToStart {
			_, err := getTransitionTimeForPodConditionWithStatus(p, v1.PodReadyToStartContainers, true)
			framework.ExpectNoError(err)
		}
```

Reply:
upadted